### DISCUSSION TITLE
Grouping ingredients that are the same, but have different names

### DISCUSSION THREAD

**Discussion by LukeRawlo** [mealie-recipes/mealie#1852] [Feature Request]
Not super sure if this has been discussed already, but it would be nice to group two ingredients that are the same thing on the shopping list. For example, one recipe calls for "4 medium egg", and another calls for "5 medium eggs, free range". If I add both of these recipes to my shopping list, it creates two entries for each one, which can become confusing when doing the shopping. I've also noticed that if two items are identical in naming, it won't consolidate them into one item. If i name the ingredients "4 eggs" on one recipe, and "4 eggs" on another recipe, it still adds 2 lots of 4 eggs

Is it possible to create a consolidation function that processes new list items when they are added, or even via a button press, that will take ingredients that are the same, and categorise them into one? It could be difficult, it may require some form of NLP / classification algorithm using a form of AI, however it would be nice to just display "9 medium eggs", rather than two sets of items.

**Comment by RealFoxie** [comment]
I don't think this is done yet, but if you use the "enable ingredient amounts" and put in your amount, unit and ingredient, it should be quite easy to merge it all together in the program. No NLP would be needed then and the solution would be a lot easier to implementation. It would not work well together with recipes that have ingredients inputted as pure text though. But I don't think such a consolidate would be possible anyway.

If it is fully automatic, it might make mistakes you don't want. How should it know you don't care about the difference of free range eggs or not? Or that you care if it are medium or large eggs? Should it consolidate those too? It could ask confirmation for consolidates, but then it becomes another chore that doesn't really help a lot with avoiding extra work.

**Comment by tit1** [comment]
I wanted to add a simmular suggestion for ingredient phrasing. Basically an ingredient dictionary of sorts that would allow you to enter alternative names, but default to the 'Main' ingredient.
for example

All-purpose flour --
     -- flour
     -- bleached flour

another example:

Granulated sugar --
     -- white sugar
     -- sugar
     -- table suagr

**Reply by terrapinlogo** [reply]
#1242 should work for this use case when it gets implemented

**Comment by michael-genson** [comment]
The shopping list will auto-combine ingredients now, and in the case where units are different it will at least group them together (so they're near each other on the shopping list)

**Reply by jjpeet** [reply]
Hi @michael-genson  I'm a brand new user of Mealie, so appologies for the dumb question. I've created/imported a bunch of recipes, and added them to the meal planner.  Can I now:

1. Export the ingredients of - say - the next 7 days of planned meals into a list? I can only seem to be able to do it one day at a time. Is that right? 
2. When in the list, like ingredients (1 tbsp olive oil & 2 tbsp olive oil) end up in the list as a single item (3 tbsp olive oil)?

I'm using version v2.4.2 (docker download a few weeks ago)

Thx
JP

**Reply by michael-genson** [reply]
Currently I think you can only do one day at a time, though we'd welcome a PR to change that!
List items will auto-combine if your foods are parsed; more on that here: https://docs.mealie.io/documentation/getting-started/faq/#how-do-i-enable-smart-ingredient-handling

**Reply by jjpeet** [reply]
Thanks @michael-genson thats works great. I also managed to get it linked to OpenAI, so the parsing works pretty well too. 
In relation to a PR, I'd like to give it a go, but given my lack of exp, it will take a while!
Can you point me in the direction of any resources that will guide me through setting up an appropriate dev env for mealie + software architecture diagrams?
Thx
JP

**Reply by michael-genson** [reply]
This should help you get started!

https://docs.mealie.io/contributors/developers-guide/starting-dev-server/
